pkg/aspect/configError.go

```diff
@@ -77,5 +77,19 @@ func (e ConfigError) Error() string {
 }
 
 func (e *ConfigErrors) Error() string {
+	if e == nil || e.Multi == nil {
+		return ""
+	}
 	return e.Multi.Error()
 }
+
+// Extend joins 2 configErrors together.
+func (e *ConfigErrors) Extend(ee *ConfigErrors) *ConfigErrors {
```

Comment:
This can be addressed out of the scope of this PR, so this should not block.

Aren't examples enough help? I'm not convinced that we should have anything other than error as the error type in our method signatures.  We can add helpers, etc. that check error types for behavior in the dispatch, etc. Use opaque errors also gives us more flexibility down the road, if we need to refactor.